pipeline/plugins/includes/tasks.py

```diff
@@ -0,0 +1,177 @@
+from airflow.decorators import task, task_group
+from airflow.providers.http.sensors.http import HttpSensor
+from airflow.providers.ssh.hooks.ssh import SSHHook
+
+from urllib.error import HTTPError
+import os
+
+from harvester import utils
+from operators.custom_operators import (
+    SaveMetsSFTPOperator,
+    SaveAltosSFTPOperator,
+)
+
+
+@task.branch(task_id="check_if_download_should_begin")
+def check_if_download_should_begin(set_id, binding_base_path, http_conn_id):
+    """
+    Check if API is responding and if there are new bindings to download.
+    If not, cancel pipeline.
+    """
+    api_ok = HttpSensor(
+        task_id="http_sensor", http_conn_id=http_conn_id, endpoint="/"
+    ).poke(context={})
+
+    bindings = utils.read_bindings(binding_base_path, set_id)
+
+    if not bindings:
+        print("No new bindings after previous download.")
+        return "cancel_pipeline"
+    if not api_ok:
+        raise HTTPError("NLF API is not responding.")
+    else:
+        return "begin_download"
+
+
+@task_group(group_id="download_set")
+def download_set(
+    set_id,
+    api,
+    ssh_conn_id,
+    initial_download,
+    image_split_dir,
+    binding_base_path,
+    base_path,
+    tmpdir,
+):
+
+    bindings = utils.read_bindings(binding_base_path, set_id)
+
+    if initial_download:
+        image_split = utils.assign_bindings_to_images(bindings, 150)
+        utils.save_image_split(image_split, image_split_dir, set_id)
+
+    else:
+        image_split = utils.assign_update_bindings_to_images(
+            bindings, image_split_dir / f"{set_id}_images.json"
+        )
+
+    image_downloads = []
+
+    for image in image_split:
+        if image["bindings"]:
+
+            @task_group(group_id=f"download_image_{set_id}_{image['prefix']}")
+            def download_image(image):
+                @task(task_id=f"prepare_download_location_{set_id}_{image['prefix']}")
+                def prepare_download_location(image):
+                    """
+                    Create an empty directory for image contents or extract an existing disk image.
+                    """
+
+                    image_base_name = f"{set_id}_{image['prefix']}".rstrip("_")
```

Comment:
Ah, now I see. This required an embarrassing amount of thinking though, and the case is not covered by tests, so I'm afraid [someone](https://github.com/aajarven) will end up breaking this if there's need for changes. I suggest we add a test or/and make this more explicit with an `if` along the lines of
```
if prefix:
    name = f"{id}_{prefix}"
else:
    name = {id}
```

Reply:
@helhotti you have a really elegant way of coding, but I again agree with @aajarven that the less elegantlooking if clause will work better with understanding the code in 3 months and for less Python-savy folks like me.

Reply:
I'm gonna need to use the rstrip anyway when setting the task IDs to avoid the trailing underscore (no way to if else those). Is it really better to then have an if statement when determining the image name? We would then have a mix of both approaches. TBH I kind of hate the if else for this, but if majority disagrees then I will do it.

Reply:
Okay, I caved, and also eliminated repetition by defining `image_base_name` in the beginning of the task group and referencing that throughout the task group in IDs etc.